kubernetes/apiserver/main.tf

```diff
@@ -25,12 +26,7 @@ resource "tls_cert_request" "apiserver" {
     common_name = "kube-apiserver"
   }
 
-  dns_names = [
-    "kubernetes",
-    "kubernetes.default",
-    "kubernetes.default.svc",
-    "kubernetes.default.svc.cluster.local"
-  ]
+  dns_names    = ["${compact(var.dns_names)}", "kubernetes", "kubernetes.default", "kubernetes.default.svc", "kubernetes.default.svc.cluster.local"]
```

Comment:
had to use compact() here otherwise it joined all together in a string.

Wonder if it would be better to set the variable default to a comma separated string and use split() here instead?

The downside is that when you inject a variable to override the dns_names you would also have to pass the default options as well